### PR TITLE
Rename --k8s-exclude-resource to --k8s-unsafe-exclude-resource

### PR DESCRIPTION
**Pull request by 2opremio**
Also, add a troubleshooting section explaning its implications.

Followup from #2749

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -149,3 +149,39 @@ releases (including automated upgrades) it has applied, and that only
...
 tag.
+
+### Flux fails with an error log similar to _couldn't get resource list for example.com/version: the server is currently unable to handle the request_
+
+This means your Kubernetes cluster fails to respond to list queries
+for resources in _example.com/version_.
+
+If the error is transient, Flux will work once the error recedes.
+
+However, the error won't normally go away by itself since it''' normally
```
Comment (stefanprodan) [inline docs/troubleshooting.md]:
```suggestion
However, the error won't go away by itself since it's normally
```

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -149,3 +149,38 @@ releases (including automated upgrades) it has applied, and that only
...
+for resources in _example.com/version_.
+
+If the error is transient, Flux will work once the error recedes.
+
+However, the error won't normally go away by itself since it's normally
+caused by a misconfiguration of your cluster.
+
+For instance, you can run into this problem when a
+[Kubernetes Webhook server](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+is removed without removing its Webhook definition.
```
Comment (stefanprodan) [inline docs/troubleshooting.md]: Please add:
Or when a custom resource definition is not available due to `FailedDiscoveryCheck` error.

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -149,3 +149,38 @@ releases (including automated upgrades) it has applied, and that only
...
 tag.
+
+### Flux fails with an error log similar to _couldn't get resource list for example.com/version: the server is currently unable to handle the request_
+
+This means your Kubernetes cluster fails to respond to list queries
+for resources in _example.com/version_.
+
+If the error is transient, Flux will work once the error recedes.
+
+However, the error won't normally go away by itself since it's normally
```
Comment (hiddeco) [inline docs/troubleshooting.md]: I would rephrase this to

> normally go away since it’s most of the time caused by [...]
